docs/registry/create/_index.md

```diff
@@ -634,7 +798,106 @@ func mainWithArgs(ctx context.Context, args []string, logger logging.Logger) (er
+
+<details>
+  <summary>Click to view sample code for <file>main.cpp</file></summary>
+
+```cpp {class="line-numbers linkable-line-numbers"}
+#include <memory>
+#include <signal.h>
+
+#include <boost/log/trivial.hpp>
+#include <grpcpp/grpcpp.h>
+#include <grpcpp/server_context.h>
+
+#include <viam/api/common/v1/common.grpc.pb.h>
+#include <viam/api/component/generic/v1/generic.grpc.pb.h>
+#include <viam/api/robot/v1/robot.pb.h>
+
+#include <viam/sdk/components/base/base.hpp>
+#include <viam/sdk/components/component.hpp>
+#include <viam/sdk/config/resource.hpp>
+#include <viam/sdk/module/module.hpp>
+#include <viam/sdk/module/service.hpp>
+#include <viam/sdk/registry/registry.hpp>
+#include <viam/sdk/resource/resource.hpp>
+#include <viam/sdk/rpc/dial.hpp>
+#include <viam/sdk/rpc/server.hpp>
+
+// NOTE: You must update the following line to import your local "my_base.hpp" header file:
+#include "my_base.hpp"
+
+using namespace viam::sdk;
+
+int main(int argc, char** argv) {
+    if (argc < 2) {
+        std::cerr << "Need socket path as command line argument" << std::endl;
+        return EXIT_FAILURE;
+    }
+    std::string socket_addr = argv[1];
+
+    // Use set_logger_severity_from_args to set the boost trivial logger's
+    // severity depending on commandline arguments.
+    set_logger_severity_from_args(argc, argv);
+    BOOST_LOG_TRIVIAL(debug) << "Starting module with debug level logging";
+
+    // C++ modules must handle SIGINT and SIGTERM. You can use the SignalManager
+    // class and its wait method to handle the correct signals.
+    SignalManager signals;
+
+    // Here is where we define your new model's colon-delimited-triplet (acme:demo:mybase)
+    // acme = namespace, demo = repo-name, mybase = model name.
+    API base_api = Base::static_api();
+    Model mybase_model("acme", "demo", "mybase");
+
+    std::shared_ptr<ModelRegistration> mybase_mr = std::make_shared<ModelRegistration>(
+        ResourceType("Base"),
+        base_api,
+        mybase_model,
+        [](Dependencies deps, ResourceConfig cfg) { return std::make_unique<MyBase>(deps, cfg); },
+        MyBase::validate);
+
+    Registry::register_model(mybase_mr);
+
+    // The `ModuleService_` must outlive the Server, so the declaration order
+    // here matters.
+    auto my_mod = std::make_shared<ModuleService_>(socket_addr);
+    auto server = std::make_shared<Server>();
+
+    my_mod->add_model_from_registry(server, base_api, mybase_model);
+    my_mod->start(server);
+    BOOST_LOG_TRIVIAL(info) << "Complex example module listening on " << socket_addr;
+
+    server->start();
+    int sig = 0;
+    auto result = signals.wait(&sig);
+    server->shutdown();
+    return EXIT_SUCCESS;
+};
+```
+
+</details>
+
+{{% alert title="Important" color="note" %}}
+
+You must define all functions belonging to a built-in resource subtype's API if defining a new model.
+Otherwise, the class will not instantiate.
```

Comment:
Is this true for C++, like it is for Python and Go?

If so, is the next line about `throw`-ing a `runtime_error` correct?

If not, what notice should be here, if any?

Reply:
This is a good question. What do we mean by `a built-in resource subtype's API`? More specifically, what is the source of truth for this? 

If we look at the header files for a built-in resource for our source of truth, there are definitely APIs that don't need to be re-implemented (see, e.g., `resource_registration()` and `dynamic_api()`. What a user _does_ need to define are all the pure virtual methods. 

If we look at what the definition of, say, a `camera` is based on `camera.proto`, then yes, you need to define all the functions.

On the below point, `throw`ing a `runtime_error` is a totally valid way of avoiding putting in an actual implementation.

Reply:
Thank you! I think we mean "camera (et al) API", which does indeed sounds like the `hpp` file definitions, like `camera.hpp`.

Ive clarified this blurb with your guidance, specifically around "implement this one but not that one."